front/lib/api/assistant/actions/process.ts

```diff
@@ -0,0 +1,416 @@
+  }
+
+  const spec = await processActionSpecification(actionConfiguration);
+  return new Ok(spec);
+}
+
+/**
+ * Action rendering.
+ */
+
+// Internal interface for the retrieval and rendering of a Process action. This should not be used
+// outside of api/assistant. We allow a ModelId interface here because we don't have `sId` on
+// actions (the `sId` is on the `Message` object linked to the `UserMessage` parent of this action).
+export async function renderProcessActionByModelId(
+  id: ModelId
+): Promise<ProcessActionType> {
+  const action = await AgentProcessAction.findByPk(id);
+  if (!action) {
+    throw new Error(`No Process action found with id ${id}`);
+  }
+
+  let relativeTimeFrame: TimeFrame | null = null;
+  if (action.relativeTimeFrameDuration && action.relativeTimeFrameUnit) {
+    relativeTimeFrame = {
+      duration: action.relativeTimeFrameDuration,
+      unit: action.relativeTimeFrameUnit,
+    };
+  }
+
+  return {
+    id: action.id,
+    type: "process_action",
+    params: {
+      relativeTimeFrame,
+    },
+    schema: action.schema,
+    outputs: action.outputs,
+  };
+}
+
+/**
+ * Action execution.
+ */
+
+// This method is in charge of running the retrieval and creating an AgentProcessAction object in
+// the database. It does not create any generic model related to the conversation. It is possible
+// for an AgentProcessAction to be stored (once the query params are infered) but for its execution
+// to fail, in which case an error event will be emitted and the AgentProcessAction won't have any
+// outputs associated. The error is expected to be stored by the caller on the parent agent message.
+export async function* runProcess(
+  auth: Authenticator,
+  {
+    configuration,
+    actionConfiguration,
+    conversation,
+    userMessage,
+    agentMessage,
+    rawInputs,
+  }: {
+    configuration: AgentConfigurationType;
+    actionConfiguration: ProcessConfigurationType;
+    conversation: ConversationType;
+    userMessage: UserMessageType;
+    agentMessage: AgentMessageType;
+    rawInputs: Record<string, string | boolean | number>;
+  }
+): AsyncGenerator<
+  ProcessParamsEvent | ProcessSuccessEvent | ProcessErrorEvent,
+  void
+> {
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Unexpected unauthenticated call to `process`");
+  }
+
+  let relativeTimeFrame: TimeFrame | null = null;
+
+  if (
+    actionConfiguration.relativeTimeFrame !== "none" &&
+    actionConfiguration.relativeTimeFrame !== "auto"
+  ) {
+    relativeTimeFrame = actionConfiguration.relativeTimeFrame;
+  }
+
+  if (actionConfiguration.relativeTimeFrame === "auto") {
+    if (
+      rawInputs.relativeTimeFrame &&
+      typeof rawInputs.relativeTimeFrame === "string"
+    ) {
+      relativeTimeFrame = parseTimeFrame(rawInputs.relativeTimeFrame);
+    }
+  }
+
+  const model = configuration.generation?.model;
+
+  let contextSize = 16384;
```

Comment:
No magical variable deep in the code 🙃.

Reply:
This is dead code to please the type system will be removed as model info is moved up to agentConfiguration